typescript/packages/common-ui/src/components/text-input.ts

```diff
@@ -0,0 +1,13 @@
+import { view } from "../hyperscript/render.js";
+import { eventProps } from "../hyperscript/schema-helpers.js";
+
+export const textInput = view(
+  "input",
+  {
+    ...eventProps(),
+    id: { type: "string" },
+    value: { type: "string" },
+    placeholder: { type: "string" },
+  },
+  { type: "text" }
```

Comment:
Rather than adding a third argument to vnode/render, if we want specialized factories, I would suggest something like:

```ts
export const textInput = (props: Props) => input({...props, type: 'text'})
```